### PR TITLE
PHOENIX-7423 End2EndTestDriver doesn't compile with latest hbase-thir…

### PR DESCRIPTION
**Pull request by stoty**
…dparty

### CODE REVIEW

**Review by virajjasani — APPROVED** [review]

**Comment by virajjasani** [comment]
This is not working for BackwardCompatibilityIT

**Comment by virajjasani** [comment]
Well, its not this Jira but PHOENIX-7418 seems to have broken many tests for HBase 2.4 and 2.5 profiles. Here is the build results: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-mulitbranch/job/master/749/

**Comment by virajjasani** [comment]
Same is the case on 5.2 https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-mulitbranch/job/5.2/66/
We will have to fix this first.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/End2EndTestDriver.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ protected void addOptions() {
 
     @Override
     protected void processOptions(CommandLine cmd) {
-      String testFilterString = cmd.getOptionValue(SHORT_REGEX_ARG, null);
+      String testFilterString = cmd.getOptionValue(SHORT_REGEX_ARG, (String)null);
```
Comment (NihalJain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/End2EndTestDriver.java:120]: we could remove the 2nd argument altogether, as the default value is anyways null. https://commons.apache.org/proper/commons-cli/apidocs/org/apache/commons/cli/CommandLine.html#getOptionValue(char)

Comment (stoty) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/End2EndTestDriver.java:120]: Yes, that's more elegant.

Comment (stoty) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/End2EndTestDriver.java]: Removed explcit default.

**Comment by stoty** [comment]
This is not the right way to fix this.

**Comment by stoty** [comment]
The proper fix is #2003

**Pull request closed without merging**